gen.go

```diff
@@ -26,7 +26,36 @@ var (
 	deferredType = reflect.TypeOf(Deferred{})
 )
 
-func doTemplate(w io.Writer, info interface{}, templ string) error {
+// Gen is a configurable code generator for CBOR types. Use this instead of the convenience
+// functions to have more control over the generated code.
+type Gen struct {
+	MaxArrayLength  int // Default: 8192 (MaxLength)
+	MaxByteLength   int // Default: 2<<20 (ByteArrayMaxLen)
+	MaxStringLength int // Default: 8192 (MaxLength)
```

Comment:
Was this actually 8192 before? That's... ok, I guess. But strange.

Reply:
yeah, strange, but also not heavily used in filecoin, so not surprising that it hasn't come up before

Reply:
yeah, was just some number that got picked as 'probably big enough for a default'